### ISSUE TITLE
Test failure on Mac: //tensorflow/core:common_runtime_gpu_gpu_allocator_retry_test

### ISSUE THREAD

**Issue by caisq** [tensorflow/tensorflow#965]
One of the C++ unit test three method in gpu_allocator_retry_test.cc failed: 
**//tensorflow/core:common_runtime_gpu_gpu_allocator_retry_test**

See below for detailed error log:

```
INFO: From Testing //tensorflow/core:common_runtime_gpu_gpu_allocator_retry_test:
==================== Test output for //tensorflow/core:common_runtime_gpu_gpu_allocator_retry_test:
Running main() from test_main.cc
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from GPUAllocatorRetryTest
[ RUN      ] GPUAllocatorRetryTest.RetrySuccess
I tensorflow/core/common_runtime/gpu/gpu_allocator_retry_test.cc:136] Consumer 0 is 1090
I tensorflow/core/common_runtime/gpu/gpu_allocator_retry_test.cc:136] Consumer 1 is 1090
I tensorflow/core/common_runtime/gpu/gpu_allocator_retry_test.cc:136] Consumer 2 is 1089
[       OK ] GPUAllocatorRetryTest.RetrySuccess (55 ms)
[ RUN      ] GPUAllocatorRetryTest.NoRetryFail
I tensorflow/core/common_runtime/gpu/gpu_allocator_retry_test.cc:160] Consumer 0 is 534506
I tensorflow/core/common_runtime/gpu/gpu_allocator_retry_test.cc:160] Consumer 1 is 534511
I tensorflow/core/common_runtime/gpu/gpu_allocator_retry_test.cc:160] Consumer 2 is 534504
**tensorflow/core/common_runtime/gpu/gpu_allocator_retry_test.cc:164: Failure
Value of: has_failed_
  Actual: false
Expected: true**
[  FAILED  ] GPUAllocatorRetryTest.NoRetryFail (10811 ms)
[ RUN      ] GPUAllocatorRetryTest.RetryInsufficientFail
I tensorflow/core/common_runtime/gpu/gpu_allocator_retry_test.cc:183] Consumer 0 is 23
I tensorflow/core/common_runtime/gpu/gpu_allocator_retry_test.cc:183] Consumer 1 is 0
I tensorflow/core/common_runtime/gpu/gpu_allocator_retry_test.cc:183] Consumer 2 is 0
[       OK ] GPUAllocatorRetryTest.RetryInsufficientFail (10001 ms)
[----------] 3 tests from GPUAllocatorRetryTest (20867 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 1 test case ran. (20867 ms total)
[  PASSED  ] 2 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] GPUAllocatorRetryTest.NoRetryFail

 1 FAILED TEST
```

**Comment by caisq** [comment]
Machine machine details:
OS: "Darwin", 
Kernel: "15.3.0", 
Architecture: "i386", 
Bazel_version: "Build label: 0.1.4-homebrew", 
Java_version: "1.8.0_65", 
Python_version: "2.7.10", 
gpp_version: "Apple LLVM version 7.0.2 (clang-700.1.81)"

**Comment by vrv** [comment]
FWIW I am unable to reproduce this failure on my MBP. Same machine details above except my kernel version is 15.2.0.

**Comment by vrv** [comment]
Interestingly, I upgraded my kernel to 15.3.0 and it failed 1 out of 10 times with the above error.

**Comment by vrv** [comment]
For expediency, should we just disable this test on Mac for now?

1) Most macs don't have GPUs that TensorFlow supports right now
2) It's not clear that gpu_retry_allocator has that much of a long life :)

Thoughts @martinwicke and @zheng-xq ?

**Comment by martinwicke** [comment]
I support whatever makes the testing work. We're not officially supporting
GPU on Mac, so not sure this needs to work.

**Comment by vrv** [comment]
@caisq: can we disable this test for the mac suite and call it a day then?

**Comment by caisq** [comment]
@vrv For the PIP install-test, it is to blacklist a test. Unfortunately this is not a PIP install-test. It is a C++ unit test. Is there an easy way to disable this C++ unit test depending on the OS in Mac i Bazel, without changing the source code?

**Comment by martinwicke** [comment]
@vrv If run without --cuda, this test is useless and shouldn't run at all,
correct? And we don't build with --cuda on Mac. Or at least we shouldn't.

**Comment by vrv** [comment]
The test doesn't require a GPU -- it's a test of a module that is used by the GPU allocator code.

**Comment by vrv** [comment]
(and I am not sure how to disable this only on mac yet.. I can look into it)

**Comment by martinwicke** [comment]
I guess my point is, if we're building without GPU support, how useful is
this test anyway -- shoudn't it be disabled whenever cuda support is off?

And if so, how would we do that? The py_cuda_tests only enable cuda, they
don't disable tests without cuda.

**Comment by vrv** [comment]
honestly, the easiest way is just to put 

```
#if GOOGLE_CUDA
``` 
around the test suite.  I can do this later today.

**Comment by martinwicke** [comment]
Do we have a similar thing for python tests? If not, we should probably
make one.

**Comment by vrv** [comment]
Turns out using #if GOOGLE_CUDA is a bad idea.

Newest proposal:
add 'tags=["nomac"]' to the test, and make jenkins Mac build invoke bazel using:
```
bazel test --test_tag_filters=-nomac
```

@caisq: does this sound reasonable?

**Comment by caisq** [comment]
This sounds reasonable to me, @vrv. However, one thing I don't understand is why the most recent several mac builds show the test passed... Did you already change some related things? See:
http://ci.tensorflow.org/job/tensorflow-master-mac/139/
http://ci.tensorflow.org/job/tensorflow-master-mac/140/
http://ci.tensorflow.org/job/tensorflow-master-mac/141/

**Comment by vrv** [comment]
I think the test is just flaky, and other recent changes might have made it less flaky.  If it doesn't fail again, maybe there's nothing to do for now.

**Comment by caisq** [comment]
Agreed, @vrv. The most recent one failed again. http://ci.tensorflow.org/job/tensorflow-master-mac/142/console

Please implement the 'tags=["nomac"]' solution.

**Comment by vrv** [comment]
"Fixed" in 155cb0ff84e9ece3bee0f06eb3bf5f92d5ca15ac
